### PR TITLE
feat: add user guided states on swap

### PR DESCRIPTION
**Pull request by luizstacio**
- Fixes price calculation per token on the preview. Ref.: #126 
- Add validation states to guide users throw the flow. Closes: #132

### CODE REVIEW

File: packages/app/src/pages/SwapPage/index.tsx [hunk 1]
```diff
@@ -1,36 +1,74 @@
...
+  if (!swapState?.to || !swapState?.from)
+    return ValidationStateEnum.SelectToken;
+  if (!swapState?.amount) {
+    return ValidationStateEnum.EnterAmount;
+  }
+  if (!swapState.hasBalance) {
+    return ValidationStateEnum.InsufficientBalance;
+  }
+  if (!hasLiquidity || isSwayInfinity(previewAmount))
+    return ValidationStateEnum.InsufficientLiquidity;
```
Comment (LuizAsFight) [inline packages/app/src/pages/SwapPage/index.tsx:62]: some `if`s have `brackets` and others don't. Either add brackets to all, or remove from all and keep only 1 line

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

File: packages/app/src/pages/SwapPage/index.tsx (hunk 1 above)
Comment (luizstacio) [reply packages/app/src/pages/SwapPage/index.tsx:62]: it happens because of the auto format we have

**Comment by LuizAsFight** [comment]
When changing `to` amount (DAI), it should not validate the balance. I think for a swap only `from` balance should be validated


https://user-images.githubusercontent.com/8636507/169871271-5cea673c-f3cc-4591-917a-cb91aa23347b.mov

**Comment by LuizAsFight** [comment]
@luizstacio what's the use case to get the ` Insufficient liquidity` feedback ?

**Comment by luizstacio** [comment]
> @luizstacio what's the use case to get the ` Insufficient liquidity` feedback ?

You need to ask for one amount bigger them the reserves

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
bug with `to` balance amount validation still not fixed.

**Pull request merged**